File: corespringsecurity/src/main/java/io/security/corespringsecurity/controller/user/UserController.java
package io.security.corespringsecurity.controller.user;


import io.security.corespringsecurity.domain.AccountDto;
import io.security.corespringsecurity.repository.UserRepository;
import io.security.corespringsecurity.service.UserService;
import io.security.corespringsecurity.service.impl.UserServiceImpl;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;

@Controller
@RequiredArgsConstructor
@RequestMapping("/")
@Slf4j
public class UserController {
	final private UserService userService;

	@GetMapping(value="/mypage")
	public String myPage() throws Exception {
		return "user/mypage";
	}

	@GetMapping("/users")
	public String createUser() {
		return "user/login/register";
	}

	@PostMapping("/users")
	public String register(@RequestBody AccountDto account) {
		System.out.println("hello");
		log.info("hello");
		userService.createUser(account);
		return "redirect:/";
	}
}

File: a/src/main/kotlin/a/a/a/controller/UserController.kt
package a.a.a.controller

import a.a.a.domain.Role
import a.a.a.domain.User
import a.a.a.dto.RoleToUserDto
import a.a.a.service.UserServiceImpl
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.ResponseBody
import org.springframework.web.bind.annotation.RestController

@RestController
class UserController (
    private val userService: UserServiceImpl
) {

    @GetMapping("/users")
    fun getUsers(): List<User> {
        return userService.findAll();
    }

    @PostMapping("/user/save")
    fun saveUser(@RequestBody user: User): User {
        return userService.saveUser(user)
    }

    @PostMapping("role/save")
    fun saveRole(@RequestBody role: Role): Role {
        return userService.saveRole(role)
    }

    @PostMapping("role/addtouser")
    fun addToUser(@RequestBody roleToUser: RoleToUserDto) {
        userService.addRoleToUser(roleToUser)
    }
}